Build fails with CMake 4+

Trying to build the library with CMake 4+ (4.0.2 in my case) fails with:

```
CMake Error at CMakeLists.txt:2 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
```

When passing `-DCMAKE_POLICY_VERSION_MINIMUM=3.5`, a number of warnings is emitted, followed by 20+ identical errors:

```
CMake Error at src/python/pycopy.cmake:19 (add_custom_command):
  No TARGET 'py3-copy' has been created in this directory.
Call Stack (most recent call first):
  src/python/src/CMakeLists.txt:57 (include)
```

I have cobbled together a patch to get the build passing, but my understanding of what goes on in `src/python` in regards
to the custom commands is basically zero, so it would likely need to be updated.

--- a/README.md
+++ b/README.md
@@ -27,7 +27,7 @@ for python bindings:
 
 for C library tests:
 
-*   check http://check.sourceforge.net/
+*   check https://github.com/libcheck/check
 
 for documentation build:
 
@@ -128,4 +128,3 @@ Here's the most direct way to get your work merged into the project.
 
 1. Push the branch to your fork
 1. Send a pull request for your branch
-

--- a/libcomps/CMakeLists.txt
+++ b/libcomps/CMakeLists.txt
@@ -1,5 +1,8 @@
+cmake_minimum_required (VERSION 3.10)
 project(libcomps C)
-cmake_minimum_required (VERSION 2.8.10)
+
+cmake_policy(SET CMP0148 OLD)
+cmake_policy(SET CMP0175 NEW)
 
 include (GNUInstallDirs)
 include (${CMAKE_ROOT}/Modules/CheckFunctionExists.cmake)
@@ -32,7 +35,7 @@ include_directories("${PROJECT_SOURCE_DIR}/src")
 #include_directories("${PROJECT_SOURCE_DIR}/src/libcomps")
 
 if (ENABLE_TESTS)
-  find_library(CHECK_LIBRARY NAMES check)
+  find_library(CHECK_LIBRARY REQUIRED NAMES check)
 endif()
 find_library(EXPAT_LIBRARY NAMES expat)
 

--- a/libcomps/src/python/docs/CMakeLists.txt
+++ b/libcomps/src/python/docs/CMakeLists.txt
@@ -26,7 +26,8 @@ add_dependencies(pydocs pycomps)
 include(../pycopy.cmake)
 add_custom_command(TARGET pydocs PRE_BUILD COMMAND set -E $ENV{LD_LIBRARY_PATH} "${LIBCOMPS_OUT}:$ENV{LD_LIBRARY_PATH}")
 
-add_custom_command(TARGET pydocs COMMAND ${PYTHON_EXECUTABLE} ${SPHINX_EXECUTABLE} -E -b html
+add_custom_command(TARGET pydocs POST_BUILD
+                                 COMMAND ${PYTHON_EXECUTABLE} ${SPHINX_EXECUTABLE} -E -b html
                                   "${CMAKE_CURRENT_SOURCE_DIR}/doc-sources/"
                                   "${CMAKE_CURRENT_BINARY_DIR}/html/"
                                  COMMENT "LDLP $ENV{LD_LIBRARY_PATH}")

--- a/libcomps/src/python/pycopy.cmake
+++ b/libcomps/src/python/pycopy.cmake
@@ -6,9 +6,10 @@ math (EXPR len "${len} - 1")
 
 #set(pycopy "py${pversion}-copy")
 
-#if (NOT TARGET ${pycopy})
+if (NOT TARGET ${pycopy})
+    add_custom_target(${pycopy} DEPENDS pycomps)
+endif()
 
-#add_custom_target(${pycopy} DEPENDS pycomps)
 set (pycomps_SRCDIR "${PROJECT_SOURCE_DIR}/src/python/src/")
 set (pycomps_TESTDIR "${PROJECT_SOURCE_DIR}/src/python/tests/")
 set (pycomps_LIBPATH ${PYCOMPS_LIB_PATH})#"${PROJECT_BINARY_DIR}/src/python/src/python${pversion}")
@@ -16,7 +17,7 @@ set (pycomps_LIBPATH ${PYCOMPS_LIB_PATH})#"${PROJECT_BINARY_DIR}/src/python/src/
 #add_custom_command(TARGET pycopy PRE_BUILD COMMAND ${CMAKE_COMMAND} -E
 #                    make_directory "${CP_DST}")
 
-add_custom_command(TARGET ${pycopy} COMMAND ${CMAKE_COMMAND} -E
+add_custom_command(TARGET ${pycopy} POST_BUILD COMMAND ${CMAKE_COMMAND} -E
                     make_directory ${pycomps_LIBPATH}/libcomps/comps/)
 
 foreach(x RANGE 0 ${len})

--- a/libcomps/tests/CMakeLists.txt
+++ b/libcomps/tests/CMakeLists.txt
@@ -87,7 +87,5 @@ add_custom_target(test_parse_run
                    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
                    DEPENDS test_parse
                    COMMENT "Running comps_parse test")
-target_link_libraries(test_parse_run libcomps)
-target_link_libraries(test_comps_run libcomps)
 
 add_dependencies(ctest test_comps_run test_parse_run)

